src/actions/sources.js

```diff
@@ -207,6 +208,10 @@ export function selectSource(id: string, options: SelectSourceOptions = {}) {
       [PROMISE]: (async () => {
         await dispatch(loadSourceText(source.toJS()));
         await dispatch(setOutOfScopeLocations());
+        const src = getSource(getState(), id).toJS();
+        if (src.text && !isOriginalId(src.id) && isMinified(src.id, src.text)) {
```

Comment:
hey @assafdd,
 just noticed one last tiny thing. we should do add `shouldPrettyPrint(src)`. it'll look something like this
```js
if (src.text && !isOriginalId(src.id) && (isMinified(src.id, src.text) || shouldPrettyPrint(src))) { 
```
 so we make sure all holes are covered.

Reply:
could we do `shouldPrettyPrint(src)) && isMinified(src.id, src.text)`

Reply:
That might be to much as there might be files that don't meet the `isMinified` heuristics but we might still want to pretty print if `shouldPrettyPrint(src)` is `true`

Reply:
Yes we can. I move the empty text check to isMinified util. 
Otherwise "should evaluate expressions in specific scope" in expressions.spec.js breaks.

Reply:
@bomsy I tried to play  with shouldPrettyPrint()  and its seems to be truthy on any JS source that I have checked.

Reply:
cool! it's fine, i had a scenario which i found but can't find it any more so its fine